slick/src/main/scala/slick/jdbc/GetResult.fm

```diff
@@ -53,4 +54,15 @@ object GetResult {
   }
 
   def apply[T](implicit f: (PositionedResult => T)) = new GetResult[T] { def apply(rs: PositionedResult) = f(rs) }
-}
+
+  /**
+    * only support primary constructor
```

Comment:
This needs some extra documentation, something like

> Derive a `GetResult[T]` using the primary constructor of `T`
> This will only work if `T` only has one constructor.